src/engine/strat_engine/metadata.rs

```diff
@@ -184,13 +184,14 @@ impl StaticHeader {
 
 
     /// Read data from f into buffer of length _BDA_STATIC_HDR_SIZE.
-    fn read_into_buf<F>(f: &mut F) -> EngineResult<[u8; _BDA_STATIC_HDR_SIZE]>
+    /// Return a newly constructed buffer and the number of bytes read.
+    fn read_into_buf<F>(f: &mut F) -> EngineResult<([u8; _BDA_STATIC_HDR_SIZE], usize)>
         where F: Read + Seek
     {
         try!(f.seek(SeekFrom::Start(0)));
         let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
-        try!(f.read(&mut buf));
-        Ok(buf)
+        let bytes = try!(f.read(&mut buf));
```

Comment:
It never had that semantics in the first place. It read all the bytes it found, left the rest of the buffer filled in with zeros, and then fired away its various discovery mechanisms.